src/main/java/core/basesyntax/service/impl/CsvGridReadService.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Grid;
+import core.basesyntax.service.GridReadService;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvGridReadService implements GridReadService {
+    private static final String SEPARATOR = ",";
+    private Grid grid;
+
+    public CsvGridReadService(File file) {
+        grid = readGrid(file);
+    }
```

Comment:
you are basically having all of the logic inside the constructor, we should avoid that.
Consider having only 1 method read with a String file path parameter, same for writer